gradio/processing_utils.py

```diff
@@ -518,3 +522,61 @@ def _scale(a, n, m, copy=True):
 
 def strip_invalid_filename_characters(filename: str) -> str:
     return "".join([char for char in filename if char.isalnum() or char in "._- "])
+
+
+def ffmpeg_installed() -> bool:
+    return shutil.which("ffmpeg") is not None
+
+
+def video_is_playable(video_filepath: str) -> bool:
+    """Determines if a video is playable in the browser.
+
+    A video is playable if it has a playable container and codec.
+        .mp4 -> h264
+        .webm -> vp9
+        .ogg -> theora
+    """
+    try:
+        container = pathlib.Path(video_filepath).suffix.lower()
+        output = subprocess.run(
+            [
+                "ffprobe",
+                "-show_format",
+                "-show_streams",
+                "-select_streams",
+                "v",
+                "-print_format",
+                "json",
+                video_filepath,
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        output = json.loads(output.stdout)
+        video_codec = output["streams"][0]["codec_name"]
+        return (container, video_codec) in [
+            (".mp4", "h264"),
+            (".ogg", "theora"),
+            (".webm", "vp9"),
+        ]
+    # If anything goes wrong, assume the video can be played to not convert downstream
+    except Exception:
```

Comment:
Nit: can we narrow the scope of this Exception?

Reply:
I got a ```video_codec= output["streams"][0]["codec_name"]
IndexError: list index out of range
 ``` 
The narrowed scope of the exception broke my application, it worked when I reverted back to `except Exception:`

Reply:
Ah yikes that's good to know. I think it would be still be better to have a scope that includes `IndexError` as well, but WDYT @freddyaboulton?

Reply:
@lazarusking sorry to hear that! I originally used a wide exception because analyzing video streams with ffmpeg felt really unpredictable. 

Can you provide a reproducer so we can dig in further? I’m curious why ffmpeg can’t find any video streams in your file, yet the video can be played in the browser if comment this out.

Reply:
[My ffmpeg demo](https://github.com/lazarusking/gradio-ffmpeg)
It basically takes a file and converts it to the chosen format, allowing you the option to play it as video, audio or just as a file. 
In the older gradio version, playing an audio as a video still works(when you toggle to video display option) but the new one doesn't find any video stream for such a case and just breaks.
It didn't account for "if anything goes wrong in this case".

Reply:
Thanks - I can put up a PR to also catch IndexError and KeyError!

Reply:
Thank you!